gutils/gimage.c

```diff
@@ -44,6 +45,9 @@ GImage *GImageCreate(enum image_type type, int32 width, int32 height) {
 	goto errorGImageCreate;
 
     gi->u.image = base;
+    base->refdata.reference = false;
+    base->refdata.filename = NULL;
+    base->refdata.hash = "0";
```

Comment:
At the very least, this should be NULL and not `"0"` otherwise you'll run into issues in all the other places that you've assumed this to be a dynamically allocated object.

But I'd also suggest just working internally with the raw representation internally (which should be a fixed length of bytes - [g_checksum_digest](https://developer.gnome.org/glib/stable/glib-Data-Checksums.html#g-checksum-get-digest)), meaning you don't need to do dynamic allocation at all.